src/components/notifications/notifications-list.tsx

```diff
@@ -0,0 +1,207 @@
+import { useGraphqlSwr } from '@/api/use-graphql-swr'
+import {
+  NotificationEvent,
+  NotificationEventPayload,
+  NotificationEventType,
+  parseNotificationEvent,
+} from '@/events/event'
+
+export function NotificationsList() {
+  const { data } = useGraphqlSwr<{
+    notifications: {
+      nodes: {
+        id: number
+        unread: boolean
+        event: NotificationEventPayload
+      }[]
+    }
+  }>({
+    query: `
+      query notifications($count: Int!, $unread: Boolean){
+        notifications(first: $count, unread: $unread) {
+          nodes {
+            id
+            unread
+            event {
+              type
+              date
+              actor {
+                id
+                username
+              }
+              object {
+                id
+              }
+              payload
+            }
+          }
+        }
+      }
+    `,
+    variables: {
+      // TODO: set number of items to show. We could add pagination in a later iteration
+      count: 20,
+      // TODO: can be true | false | undefined. If defined, it will only return notifications that have the corresponding unread state.
+      unread: undefined,
+    },
+  })
+  return (
+    <ul>
+      {data?.notifications.nodes.map((node) => {
+        const props = {
+          ...node,
+          event: parseNotificationEvent(node.event),
+        }
+        return (
+          <li key={node.id}>
+            {node.unread ? 'ungelesen' : 'gelesen'}, {node.event.date}
+            <br />
+            <Notification {...props} />
+          </li>
+        )
+      })}
+    </ul>
+  )
+}
+
+function Notification({
+  event,
+}: {
+  id: number
+  unread: boolean
+  event: NotificationEvent
+}) {
+  switch (event.type) {
+    case NotificationEventType.SetThreadState:
```

Comment:
Basic boilerplate to render the actual notifications (covers all possible event types)